internal/backup/volumesnapshotcontent_action.go

```diff
@@ -99,34 +100,36 @@ func (p *VolumeSnapshotContentBackupItemActionV2) Execute(item runtime.Unstructu
 			return nil, nil, "", nil, errors.WithStack(err)
 		}
 
-		// craft a VolumeBackupSnapshot object to be created
-		vsb := datamoverv1alpha1.VolumeSnapshotBackup{
-			ObjectMeta: metav1.ObjectMeta{
-				GenerateName: "vsb-",
-				Namespace:    snapCont.Spec.VolumeSnapshotRef.Namespace,
-				Labels: map[string]string{
-					util.BackupNameLabel: backup.Name,
-				},
-			},
-			Spec: datamoverv1alpha1.VolumeSnapshotBackupSpec{
-				VolumeSnapshotContent: corev1api.ObjectReference{
-					Name: snapCont.Name,
-				},
-				ProtectedNamespace: backup.Namespace,
-				ResticSecretRef: corev1api.LocalObjectReference{
-					Name: resticSecretName,
-				},
-			},
-		}
-
-		// check if VolumeBackupSnapshot CR exists for VSC
-		VSBExists, err := util.DoesVolumeSnapshotBackupExistForVSC(&snapCont, p.Log)
+		// check if VolumeSnapshotBackup CR exists for VolumeSnapshotContent
+		VSBExists, err := util.VSBExistsForVSC(&snapCont, p.Log)
 		if err != nil {
 			return nil, nil, "", nil, errors.WithStack(err)
 		}
 
 		// Create VSB only if does not exist for the VSC
 		if !VSBExists {
+
+			// craft a VolumeBackupSnapshot object to be created
+			vsb := datamoverv1alpha1.VolumeSnapshotBackup{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "vsb-",
+					Namespace:    snapCont.Spec.VolumeSnapshotRef.Namespace,
+					Labels: map[string]string{
+						util.BackupNameLabel:                           backup.Name,
```

Comment:
Would it be possible to add backup UID here?

backup.Name isn't going to cause any issues here but it would be helpful to be able to compare if backup CR in the cluster still applies to this VSB.